driver-examples/src/main/java/com/datastax/driver/examples/datatypes/Blobs.java

```diff
@@ -0,0 +1,251 @@
+
+    static String[] CONTACT_POINTS = {"127.0.0.1"};
+    static int PORT = 9042;
+
+    static File FILE = new File(Blobs.class.getResource("/cassandra_logo.png").getFile());
+
+    public static void main(String[] args) throws IOException {
+        Cluster cluster = null;
+        try {
+            cluster = Cluster.builder()
+                    .addContactPoints(CONTACT_POINTS).withPort(PORT)
+                    .build();
+            Session session = cluster.connect();
+
+            createSchema(session);
+            allocateAndInsert(session);
+            retrieveSimpleColumn(session);
+            retrieveMapColumn(session);
+            insertConcurrent(session);
+            insertFromAndRetrieveToFile(session);
+        } finally {
+            if (cluster != null) cluster.close();
+        }
+    }
+
+    private static void createSchema(Session session) {
+        session.execute("CREATE KEYSPACE IF NOT EXISTS examples " +
+                "WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1}");
+        session.execute("CREATE TABLE IF NOT EXISTS examples.blobs(k int PRIMARY KEY, b blob, m map<text, blob>)");
+    }
+
+    private static void allocateAndInsert(Session session) {
+        // One way to get a byte buffer is to allocate it and fill it yourself:
+        ByteBuffer buffer = ByteBuffer.allocate(16);
+        while (buffer.hasRemaining())
+            buffer.put((byte) 0xFF);
+
+        // Don't forget to flip! The driver expects a buffer that is ready for reading. That is, it will consider all
+        // the data between buffer.position() and buffer.limit().
+        // Right now we are positioned at the end because we just finished writing, so if we passed the buffer as-is it
+        // would appear to be empty:
+        assert buffer.limit() - buffer.position() == 0;
+
+        buffer.flip();
+        // Now position is back to the beginning, so the driver will see all 16 bytes.
+        assert buffer.limit() - buffer.position() == 16;
+
+        session.execute("INSERT INTO examples.blobs (k, b, m) VALUES (1, ?, ?)",
+                buffer, ImmutableMap.of("test", buffer));
+    }
+
+    private static void retrieveSimpleColumn(Session session) {
+        Row row = session.execute("SELECT b, m FROM examples.blobs WHERE k = 1").one();
+
+        ByteBuffer buffer = row.getBytes("b");
+
+        // The driver always returns buffers that are ready for reading.
+        assert buffer.limit() - buffer.position() == 16;
+
+        // One way to read from the buffer is to use absolute getters. Do NOT start reading at index 0, as the buffer
+        // might start at a different position (we'll see an example of that later).
+        for (int i = buffer.position(); i < buffer.limit(); i++) {
+            assert buffer.get(i) == (byte) 0xFF;
+        }
+
+        // Another way is to use relative getters.
+        while (buffer.hasRemaining()) {
+            assert buffer.get() == (byte) 0xFF;
```

Comment:
If you don't have assertions enabled (`-ea` vm argument) this will tight loop since buffer.get() is not evaluated.   Maybe we should change these to not use assertions?  Either that or we could document that assertions are required.